Replace black with ruff and fix import declarations and f string

Dvdandrades opened the pull request:
## Summary of the changes / Why this is an improvement
Add Ruff to dev dependencies
Remove black from dev dependencies
Update GitHub Actions CI lint job to run Ruff instead of black
Update DEVELOPMENT.md 

Run Ruff and fix imports declarations and f string

Add Ruff config in pyproject.toml with just the line-lenght. It can be improved with more rules

## Checklist

 - [x] Link to issue this PR refers to (if applicable): Fixes #16

amotl commented on pyproject.toml:
```diff
@@ -26,10 +26,10 @@ dependencies = [
     "django-stubs>=5.1.3",
     "pytest-cratedb>=0.4.0",
     "requests>=2.32.3",
+    "ruff>=0.14.10",
```
It's better like this, to not cause any behaviour drift, due to new formatting rules, and deprecations of older ones.
```suggestion
    "ruff<0.15",
```

amotl replied:
Dependabot also understands this paradigm, and will submit corresponding patches, so maintainers can either merge without ado (if nothing changes), or accompany the PR with required adjustments about code formatting rules coming from a newer minor version of ruff.

- https://github.com/crate/cratedb-toolkit/pull/535

amotl commented on uv.lock:
The Django adapter should also be treated as a library, because it is usually not installed standalone, but pulled in as a dependency. In this spirit, the lock file can be removed. Package consumers can't use it anyway.

- https://github.com/orgs/python-poetry/discussions/7847
- https://doc.rust-lang.org/cargo/faq.html#why-have-cargolock-in-version-control

Dvdandrades replied:
Thanks for the information. So, from what I understand, in cases where the project is a library, it's good practice to include the lock file in the .gitignore file or simply remove it, right?

Dvdandrades commented:
I believe all the changes have been made correctly. I've updated the branch so that ruff could fix the missing error. I've updated the ruff dependency and added the .lock file to the .gitignore to ensure consistency with other repositories.

amotl commented on tests/test_base.py:
```diff
@@ -63,7 +67,7 @@ def test_get_connection_params():
     opts["OPTIONS"] = {"verify_ssl_cert": False}
     c = DatabaseWrapper(opts).get_connection_params()
-    assert c["verify_ssl_cert"] == False
+    assert not c["verify_ssl_cert"]
```
Nit: Just in case you want to be explicitly checking for `False`. In this case, an identity match is applicable, contrary to the typical equality match.
```suggestion
    assert c["verify_ssl_cert"] is False
```

Dvdandrades replied:
Suggested change done. Thank you so much for the explanations. I've learned a lot.

amotl reviewed (approved):
Excellent, thanks. Please squash [^1] your commits before merging.

[^1]: Nit: Doesn't need to be a single commit, do it at your disposal, but thinning out would be good (e.g. one for adding ruff + formatting, and another one for removing the lock file). Thanks!
